src/analyzer/protocol/websocket/websocket.spicy

```diff
@@ -0,0 +1,120 @@
+# Copyright (c) 2024 by the Zeek Project. See LICENSE for details.
+
+module WebSocket;
+
+import spicy;
+import zeek;
+
+const OPCODE_CONTINUATION = 0x00;
+const OPCODE_TEXT         = 0x01;
+const OPCODE_BINARY       = 0x02;
+const OPCODE_CLOSE        = 0x08;
+const OPCODE_PING         = 0x09;
+const OPCODE_PONG         = 0x0a;
+
+# Are bytes copied like a string in C++, or is that ref counted and cheap
+# like it would be in Python?
```

Comment:
No need to guess here: `f` is passed as `inout` so we generate code taking it as mutable ref; `chunk` is `in` so it becomes a const ref argument. If you are ever curious, just look at the generated code, e.g., here

```ruby
module foo;

type Frame = unit {};

function unmask( inout f: Frame, chunk: bytes): bytes {
    return b"";
}
```

```console
# Disable optimizations with `-g` to avoid removing unused code.
$ spicyc -cg foo.spicy
..
static auto unmask(::hilti::rt::ValueReference<__hlt::foo::Frame>& f, const ::hilti::rt::Bytes& chunk) -> ::hilti::rt::Bytes;
..
```